skbio/io/format/genbank.py

```diff
@@ -0,0 +1,683 @@
+|Yes   |Yes   |:mod:`skbio.sequence.RNA`                                      |
++------+------+---------------------------------------------------------------+
+|Yes   |Yes   |:mod:`skbio.sequence.Protein`                                  |
++------+------+---------------------------------------------------------------+
+
+Format Specification
+--------------------
+The International Nucleotide Sequence Database Collaboration (INSDC)
+foundational initiative between the DDBJ, EMBL, and Genbank
+(http://www.insdc.org/). These organisations all use the
+same "Feature Table" layout in their plain text flat file formats.
+
+However, the header and sequence sections of an EMBL file are very
+different in layout to those produced by Genbank/DDBJ.
+
+Feature Table Documentation:
+http://www.insdc.org/files/feature_table.html
+ftp://ftp.ncbi.nih.gov/genbank/docs/FTv10_3.html
+
+Examples
+--------
+
+"""
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import range, zip
+
+import re
+import numpy as np
+import pandas as pd
+from datetime import datetime
+from functools import partial
+
+from skbio.io import create_format, GenbankFormatError
+from skbio.io.format._base import (
+    _get_nth_sequence, _line_generator, _too_many_blanks)
+from skbio.util._misc import chunk_str
+from skbio.sequence import Sequence, DNA, RNA, Protein
+
+
+genbank = create_format('genbank')
+
+# date format in locus line of genbank record
+_TIME_FORMAT = '%d-%b-%Y'
+# This list is ordered
+# used to read and write genbank file.
+_HEADERS = ['LOCUS',
+            'DEFINITION',
+            'ACCESSION',
+            'VERSION',
+            'DBSOURCE',
+            'DBLINK',
+            'KEYWORDS',
+            'SOURCE',
+            'REFERENCE',
+            'COMMENT',
+            'FEATURES',
+            'ORIGIN']
+
+
+@genbank.sniffer()
+def _genbank_sniffer(fh):
+    # check the 1st real line is a valid LOCUS line
+    if _too_many_blanks(fh, 5):
+        return False, {}
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    try:
+        _parse_locus([line])
+    except GenbankFormatError:
+        return False, {}
+    return True, {}
+
+
+@genbank.reader(None)
+def _genbank_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_genbanks(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+@genbank.reader(Sequence)
+def _genbank_to_biological_sequence(fh, seq_num=1, **kwargs):
```

Comment:
Those were old, they've since been updated in `master`. We basically forgot to rename them when we went from `BiologicalSequence` -> `Sequence`